src/HttpRequest.h

```diff
@@ -254,5 +254,13 @@ class ConnStateData;
  */
 void UpdateRequestNotes(ConnStateData *csd, HttpRequest &request, NotePairs const &notes);
 
+/// \returns a pointer to known *_port address where the request was accepted
+/// \returns nil for nil requests and when the port address info was not available
+/// cacheip and localip are useful when ALE has necessary info (cache.port->s / tcpClient->local)
+/// if cacheip or localip is nil and derive_ips is set, then it tries to derive them from request
+const Ip::Address *FindListeningPortAddress(const HttpRequest *request, const bool derive_ips, const Ip::Address *cacheip, const Ip::Address *localip);
```

Comment:
I kept it public so that it can be used (in future) in cases where ALE is not available but IPs are somehow available. But I will remove it for now, it can be moved back to .h file in future, if ever required.

> ... hiding the decision complexities from the callers, and extracting information from any available source

As you said, currently it uses only one source. Either al (with al->request internally) or just request. So 1 parameter or 2 does not make any difference. (compiler would have called appropriate overloaded function)

But I am switching to 2 parameter function. Although personally I like function overloading.